pkg/templates/latesttag/template.go

```diff
@@ -49,14 +58,27 @@ func init() {
 				var results []diagnostic.Diagnostic
 
 				for _, container := range podSpec.Containers {
-					if isInList(blockedRegexes, container.Image) {
-						results = append(results, diagnostic.Diagnostic{Message: fmt.Sprintf("The container %q is using a floating image tag, %q.", container.Name, container.Image)})
-					}
+					if !(len(blockedRegexes) == 0 && len(allowedRegexes) == 0) {
+						switch {
+						case len(blockedRegexes) > 0 && len(allowedRegexes) > 0:
+							if isInList(blockedRegexes, container.Image) {
+								results = append(results, diagnostic.Diagnostic{Message: fmt.Sprintf("The container %q is using an invalid container image, %q. Please use images that are not blocked by the `BlockList` criteria : %q", container.Name, container.Image, blockedRegexes)})
+							} else if !isInList(allowedRegexes, container.Image) {
+								results = append(results, diagnostic.Diagnostic{Message: fmt.Sprintf("The container %q is using an invalid container image, %q. Please use images that satisfies the `AllowList` criteria : %q", container.Name, container.Image, allowedRegexes)})
 
+							}
+						case len(blockedRegexes) > 0 && len(allowedRegexes) == 0:
+							if isInList(blockedRegexes, container.Image) {
+								results = append(results, diagnostic.Diagnostic{Message: fmt.Sprintf("The container %q is using an invalid container image, %q. Please use images that are not blocked by the `BlockList` criteria : %q", container.Name, container.Image, blockedRegexes)})
+							}
+						case len(blockedRegexes) == 0 && len(allowedRegexes) > 0:
+							if !isInList(allowedRegexes, container.Image) {
+								results = append(results, diagnostic.Diagnostic{Message: fmt.Sprintf("The container %q is using an invalid container image, %q. Please use images that satisfies the `AllowList` criteria : %q", container.Name, container.Image, allowedRegexes)})
+							}
+						}
+					}
```

Comment:
I think this can be simplified to:

```go
if len(blockedRegexes) > 0 && isInList(blockedRegexes, container.Image) { // ... error}
else if len(allowedRegexes) > 0 && !isInList(allowedRegexes, container.Image) { // ... error }
```

Reply:
Simplified the above block as suggested. Thanks for pointing.